# Medical Device Failure & Healthcare Financial Risk BI Project

## Notebook 01: Data Import and Initial Validation

This notebook imports the raw project datasets, checks their structure, reviews column names, identifies missing values and duplicates, and creates a starting validation summary before cleaning.

In [1]:
# ==========================================
# Notebook 01: Data Import and Initial Validation
# Medical Device Failure & Healthcare Financial Risk BI Project
# ==========================================

import pandas as pd
import numpy as no
from pathlib import Path

#Display settings
pd.set_option("display.max_columns",None)
pd.set_option("display.width",120)

print("Libraries imported successfully.")

Libraries imported successfully.


In [ ]:
# ==========================================
# Set project folder paths
# ==========================================

PROJECT_ROOT = Path(r"C:\Users\Charl\OneDrive\Documents\GitHub\Medical Device Failure & Healthcare Supply Chain Risk BI Project")



Project root: C:\Users\Charl\OneDrive\Documents\GitHub\Medical Device Failure & Healthcare Supply Chain Risk BI Project


In [8]:
# ==========================================
# Check raw data files
# ==========================================

from pathlib import Path

raw_data_path = Path(
    r"C:\Users\Charl\OneDrive\Documents\GitHub\Medical Device Failure & Healthcare Supply Chain Risk BI Project\data\raw"
)

raw_files = [
    raw_data_path / "financial_data.csv",
    raw_data_path / "Medical_Device_Failure_dataset.csv"
]

raw_files 

[WindowsPath('C:/Users/Charl/OneDrive/Documents/GitHub/Medical Device Failure & Healthcare Supply Chain Risk BI Project/data/raw/financial_data.csv'),
 WindowsPath('C:/Users/Charl/OneDrive/Documents/GitHub/Medical Device Failure & Healthcare Supply Chain Risk BI Project/data/raw/Medical_Device_Failure_dataset.csv')]

In [9]:
for file in raw_files:
    print(file.name, "exists:", file.exists())

financial_data.csv exists: True
Medical_Device_Failure_dataset.csv exists: True


In [10]:
financial_df = pd.read_csv(raw_files[0])
medical_device_df=pd.read_csv(raw_files[1])

print("Financial data shape:", financial_df.shape)
print("Medical device data shape:", medical_device_df.shape)

Financial data shape: (500, 4)
Medical device data shape: (4149, 13)


In [11]:
# ==========================================
# Preview medical device failure dataset
# ==========================================

medical_device_df.head()
medical_device_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4149 entries, 0 to 4148
Data columns (total 13 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Device_ID              4149 non-null   object 
 1   Device_Type            4149 non-null   object 
 2   Purchase_Date          4149 non-null   object 
 3   Age                    4149 non-null   int64  
 4   Manufacturer           4149 non-null   object 
 5   Model                  4149 non-null   object 
 6   Country                4149 non-null   object 
 7   Maintenance_Cost       4149 non-null   float64
 8   Downtime               4149 non-null   float64
 9   Maintenance_Frequency  4149 non-null   int64  
 10  Failure_Event_Count    4149 non-null   int64  
 11  Maintenance_Class      4149 non-null   int64  
 12  Maintenance_Report     4149 non-null   object 
dtypes: float64(2), int64(4), object(7)
memory usage: 421.5+ KB


In [12]:
# ==========================================
# Preview financial dataset
# ==========================================

financial_df.head()
financial_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Date              500 non-null    object 
 1   Expense_Category  500 non-null    object 
 2   Amount            500 non-null    float64
 3   Description       500 non-null    object 
dtypes: float64(1), object(3)
memory usage: 15.8+ KB


In [13]:
# ==========================================
# Review column names
# ==========================================

print("Device dataset columns:")

print(medical_device_df.columns.tolist())


print("\nFiniancial dataset columns")
print(financial_df.columns.tolist())

Device dataset columns:
['Device_ID', 'Device_Type', 'Purchase_Date', 'Age', 'Manufacturer', 'Model', 'Country', 'Maintenance_Cost', 'Downtime', 'Maintenance_Frequency', 'Failure_Event_Count', 'Maintenance_Class', 'Maintenance_Report']

Finiancial dataset columns
['Date', 'Expense_Category', 'Amount', 'Description']


In [15]:
# ==========================================
# Review data types
# ==========================================

print("Device dataset data types:")
print(medical_device_df.dtypes)

print("\nFiniancial dataset data types:")
print(financial_df.dtypes)




Device dataset data types:
Device_ID                 object
Device_Type               object
Purchase_Date             object
Age                        int64
Manufacturer              object
Model                     object
Country                   object
Maintenance_Cost         float64
Downtime                 float64
Maintenance_Frequency      int64
Failure_Event_Count        int64
Maintenance_Class          int64
Maintenance_Report        object
dtype: object

Finiancial dataset data types:
Date                 object
Expense_Category     object
Amount              float64
Description          object
dtype: object


In [17]:
# ==========================================
# Check missing values
# ==========================================


print("Missing values in device dataset:")
print(medical_device_df.isnull().sum())


print("\nMissing values in financial dataset:")    
print(financial_df.isnull().sum())



Missing values in device dataset:
Device_ID                0
Device_Type              0
Purchase_Date            0
Age                      0
Manufacturer             0
Model                    0
Country                  0
Maintenance_Cost         0
Downtime                 0
Maintenance_Frequency    0
Failure_Event_Count      0
Maintenance_Class        0
Maintenance_Report       0
dtype: int64

Missing values in financial dataset:
Date                0
Expense_Category    0
Amount              0
Description         0
dtype: int64


In [18]:
# ==========================================
# Check duplicate rows
# ==========================================

print("Duplicate rows in device dataset:", medical_device_df.duplicated().sum())

print("Duplicate rows in financial dataset:", financial_df.duplicated().sum())



Duplicate rows in device dataset: 0
Duplicate rows in financial dataset: 0


In [19]:
# ==========================================
# Create raw dataset summary
# ==========================================

dataset_summary = pd.DataFrame({
    "Dataset": [
        "Medical Device Failure",
        "Financial Data"
    ],
    "File_Name": [
        "Medical_Device_Failure_dataset.csv",
        "financial_data.csv"
    ],
    "Row_Count": [
       medical_device_df.shape[0],
       financial_df.shape[0]
    ],
    "Column_Count": [
        medical_device_df.shape[1],
        financial_df.shape[1]
    ],
    "Duplicate_Rows": [
        medical_device_df.duplicated().sum(),
        financial_df.duplicated().sum()
    ],
    "Total_Missing_Values": [
        medical_device_df.isnull().sum().sum(),
        financial_df.isnull().sum().sum()
    ]
})

dataset_summary

,Dataset,File_Name,Row_Count,Column_Count,Duplicate_Rows,Total_Missing_Values
0,Medical Device Failure,Medical_Device_Failure_dataset.csv,4149,13,0,0
1,Financial Data,financial_data.csv,500,4,0,0


In [20]:
# ==========================================
# Export raw dataset summary
# ==========================================

dataset_summary.to_csv(r"C:\Users\Charl\OneDrive\Documents\GitHub\Medical Device Failure & Healthcare Supply Chain Risk BI Project\data\exports\dataset_summary.csv", index=False)

print("Raw dataset summary exported successfully.")

Raw dataset summary exported successfully.


In [21]:
# ==========================================
# Basic descriptive statistics: device dataset
# ==========================================

medical_device_df.describe()


,Age,Maintenance_Cost,Downtime,Maintenance_Frequency,Failure_Event_Count,Maintenance_Class
count,4149.000000,4149.000000,4149.000000,4149.00000,4149.000000,4149.000000
mean,6.717763,8643.595669,10.621367,3.01181,1.957821,2.000000
std,3.175610,5930.038551,7.099434,1.42173,1.413328,0.816595
min,1.000000,-5747.932890,0.950068,1.00000,0.000000,1.000000
25%,4.000000,4451.775127,5.066061,2.00000,1.000000,1.000000
50%,7.000000,7213.575604,8.862348,3.00000,2.000000,2.000000
75%,9.000000,12023.912821,14.638223,4.00000,3.000000,3.000000
max,12.000000,34937.222389,34.152648,5.00000,10.000000,3.000000


In [22]:
# ==========================================
# Basic descriptive statistics: financial dataset
# ==========================================

financial_df.describe()

,Amount
count,500.000000
mean,24720.491460
std,13731.361055
min,1110.270000
25%,13213.532500
50%,24028.275000
75%,36010.815000
max,49581.160000


In [23]:
# ==========================================
# Review key categorical values
# ==========================================

print("Device Types:") 
print(medical_device_df["Device_Type"].value_counts())

print("\nManufacturers:") 
print(medical_device_df["Manufacturer"].value_counts())

print("\nMaintenance classes:")
print(medical_device_df["Maintenance_Class"].value_counts())


print("\nFinancial expense categories:") 
print(financial_df["Expense_Category"].value_counts())

print("\nFinancial descriptions:") 
print(financial_df["Description"].value_counts())


Device Types:
Device_Type
X-Ray Machine    729
Infusion Pump    705
Defibrillator    701
PET Scanner      679
MRI Scanner      673
CT Scanner       662
Name: count, dtype: int64

Manufacturers:
Manufacturer
ScanPro        729
ImagingTech    698
MedEquip       691
RescueTech     689
CardioSync     677
UltraMed       665
Name: count, dtype: int64

Maintenance classes:
Maintenance_Class
1    1383
2    1383
3    1383
Name: count, dtype: int64

Financial expense categories:
Expense_Category
Supplies     188
Staffing     167
Equipment    145
Name: count, dtype: int64

Financial descriptions:
Description
Surgeons' salaries    183
Surgical masks        172
Ventilators           145
Name: count, dtype: int64


## Initial Data Import Notes

Both raw datasets were imported successfully.

The medical device failure dataset contains 4,149 rows and 13 columns.<br>
It includes device type, purchase date, device age, manufacturer, model, <br>
country, maintenance cost, downtime, maintenance frequency, failure event count,<br>
 maintenance class, and maintenance report text.<br>

The financial dataset contains 500 rows and 4 columns. It includes expense date, expense category, amount, and description.<br>

Initial validation showed no duplicate rows and no missing values in either dataset. Date columns were imported as text and will be converted to datetime format during the cleaning phase.<br>

The financial dataset will require category validation because some expense descriptions may not match the original assigned expense category.